src/vm/eventpipe.cpp

```diff
@@ -786,6 +787,14 @@ EventPipeEventInstance *EventPipe::GetNextEvent(EventPipeSessionID sessionID)
     return pSession ? pSession->GetNextEvent() : nullptr;
 }
 
+HANDLE EventPipe::GetWaitHandle(EventPipeSessionID sessionID)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    EventPipeSession *const pSession = GetSession(sessionID);
+    return pSession ? pSession->GetWaitEvent()->GetHandleUNHOSTED() : 0;
```

Comment:
Is there a valid scenario where we expect this method to return 0? I'm not sure there is, but if so and I missed it I don't think DispatchEventsToEventListeners handles it.

Reply:
I just copied this logic from EventPipe::GetNextEvent